### ISSUE TITLE
`smtp_password` field is still valid in V3 API

### ISSUE THREAD

**Issue by samm-git** [terraform-providers/terraform-provider-mailgun#26]
Hi,

Recently `smtp_password` field on domain resource was removed. However, i found that it is still working as it should. Here is an example:

```
curl -s --user "api:${API_KEY}"   -X POST https://api.mailgun.net/v3/domains   -F name='mytest.example.com'   -F smtp_password='supersecretpassword1234'
{
  "domain": {
    "created_at": "Fri, 25 Oct 2019 16:10:29 GMT",
    "id": "5db31e7580b758a34acef1f7",
    "is_disabled": false,
    "name": "tmytest.example.com",
    "require_tls": false,
    "skip_verification": false,
    "smtp_login": "postmaster@tmytest.example.com",
    "smtp_password": "supersecretpassword1234",
    "spam_action": "disabled",
    "state": "unverified",
    "type": "custom",
    "web_prefix": "email",
    "web_scheme": "http",
    "wildcard": false
  },
  "message": "Domain has been created",
  "receiving_dns_records": [
    {
      "cached": [],
      "priority": "10",
      "record_type": "MX",
      "valid": "unknown",
      "value": "mxa.mailgun.org"
    },
    {
      "cached": [],
      "priority": "10",
      "record_type": "MX",
      "valid": "unknown",
      "value": "mxb.mailgun.org"
    }
  ],
  "sending_dns_records": [
    {
      "cached": [],
      "name": "tmytest.example.com",
      "record_type": "TXT",
      "valid": "unknown",
      "value": "v=spf1 include:mailgun.org ~all"
    },
    {
      "cached": [],
      "name": "mx._domainkey.tmytest.example.com",
      "record_type": "TXT",
      "valid": "unknown",
      "value": "k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDM7buRlrVDW58Gm3585TqwI+NyMI3WvnGtjMg6S2UILQq+vW+ZZcMY20KLw4uoMd9WBhLtzk6+AulMYiG2XiZDkvCEcaWsnB9KRJQ1VAB9+xqGEEPB9vNUHmfzPewx7Dk5kw8rG31ZFCP9kJN9fdo0ME3nuyz/cdkihYOxIWKMkwIDAQAB"
    },
    {
      "cached": [],
      "name": "email.tmytest.example.com",
      "record_type": "CNAME",
      "valid": "unknown",
      "value": "mailgun.org"
    }
  ]
}
```
I think that commit which removing smtp_password needs to be reverted.

**Comment by samm-git** [comment]
Related commits: https://github.com/terraform-providers/terraform-provider-mailgun/commit/61063ea89137ce1f45c1138337340703e072f8e5, https://github.com/terraform-providers/terraform-provider-mailgun/commit/988e35bfe7a79801a4f0aa60187fc8cbd3b9ad26#diff-1ad9f66c36cd769f8e774d9ee8be2020

P.S. problem is that call is failing with a simple passwords causing misleading message. I opened case on the mailgun side as well

**Comment by samm-git** [comment]
I contacted mailgun support and they confirmed that password still can be set from the API. Also i informed them about misleading message in case of non-strong password, they will send this information to the app team.

**Comment by wgebis** [comment]
Hi!
thanks for feedback, I'll verify this case and fix the issue. Your help is welcome too, feel free to make a PR :).  
Cheers!

**Comment by wgebis** [comment]
Confirmed, changes committed and will be released in v0.4.2 soon! Thanks @samm-git 👍
